### ISSUE TITLE
Implement scatter_softmax() within mlx.core

### ISSUE THREAD

**Issue by TristanBilot** [TristanBilot/mlx-graphs#2]
In order to implement GAT, we need a working `scatter_softmax()` function, along with its `vjp()` to compute gradients.
Such core function has to implemented in cpp within `mlx.core`, directly in the [mlx repo](https://github.com/ml-explore/mlx).

Once implemented, we can use this function from `mlx-graphs` via python binders.

Conversation about `scatter_add()` within MLX: https://github.com/ml-explore/mlx/discussions/351#discussioncomment-8035476
Scatter softmax from PyG: https://pytorch-scatter.readthedocs.io/en/1.4.0/composite/softmax.html

**Comment by TristanBilot** [comment]
PR for adding `scatter_add` vjp in mlx: https://github.com/ml-explore/mlx/pull/431
